--- jabref-2.11.1/src/main/java/net/sf/jabref/external/DroppedFileHandler.java.jgoodies
+++ jabref-2.11.1/src/main/java/net/sf/jabref/external/DroppedFileHandler.java
@@ -35,8 +35,7 @@
 import net.sf.jabref.util.Util;
 import net.sf.jabref.util.XMPUtil;
 
-import com.jgoodies.forms.builder.DefaultFormBuilder;
-import com.jgoodies.forms.layout.CellConstraints;
+import com.jgoodies.forms.builder.FormBuilder;
 import com.jgoodies.forms.layout.FormLayout;
 
 /**
@@ -87,16 +86,15 @@
 
         FormLayout layout = new FormLayout("left:15dlu,pref,pref,pref", "bottom:14pt,pref,pref,pref,pref");
         layout.setRowGroups(new int[][] {{1, 2, 3, 4, 5}});
-        DefaultFormBuilder builder = new DefaultFormBuilder(layout, optionsPanel);
-        builder.border(Borders.DIALOG);
-        CellConstraints cc = new CellConstraints();
-
-        builder.add(linkInPlace, cc.xyw(1, 1, 4));
-        builder.add(destDirLabel, cc.xyw(1, 2, 4));
-        builder.add(copyRadioButton, cc.xyw(2, 3, 3));
-        builder.add(moveRadioButton, cc.xyw(2, 4, 3));
-        builder.add(renameCheckBox, cc.xyw(2, 5, 1));
-        builder.add(renameToTextBox, cc.xyw(4, 5, 1));
+        FormBuilder builder = FormBuilder.create().layout(layout);
+         
+        builder.add(linkInPlace).xyw(1, 1, 4);
+        builder.add(destDirLabel).xyw(1, 2, 4);
+        builder.add(copyRadioButton).xyw(2, 3, 3);
+        builder.add(moveRadioButton).xyw(2, 4, 3);
+        builder.add(renameCheckBox).xyw(2, 5, 1);
+        builder.add(renameToTextBox).xyw(4, 5, 1);
+        optionsPanel.add(builder.getPanel());
 
     }
 
--- jabref-2.11.1/src/main/java/net/sf/jabref/gui/CleanUpAction.java.jgoodies
+++ jabref-2.11.1/src/main/java/net/sf/jabref/gui/CleanUpAction.java
@@ -37,8 +37,7 @@
 import net.sf.jabref.undo.NamedCompound;
 import net.sf.jabref.undo.UndoableFieldChange;
 
-import com.jgoodies.forms.builder.DefaultFormBuilder;
-import com.jgoodies.forms.layout.CellConstraints;
+import com.jgoodies.forms.builder.FormBuilder;
 import com.jgoodies.forms.layout.FormLayout;
 import net.sf.jabref.util.DOIUtil;
 import net.sf.jabref.util.FileUtil;
@@ -140,25 +139,24 @@
         retrieveSettings();
 
         FormLayout layout = new FormLayout("left:15dlu,pref:grow", "pref, pref, pref, pref, pref, pref, pref, pref, pref, pref, pref, pref, pref, pref, pref");
-        DefaultFormBuilder builder = new DefaultFormBuilder(layout, optionsPanel);
-        builder.border(Borders.DIALOG);
-        CellConstraints cc = new CellConstraints();
-        builder.add(cleanUpHTML, cc.xyw(1, 1, 2));
-        builder.add(cleanUpUnicode, cc.xyw(1, 2, 2));
-        builder.add(cleanUpCase, cc.xyw(1, 3, 2));
-        builder.add(cleanUpLaTeX, cc.xyw(1, 4, 2));
-        builder.add(cleanUpUnits, cc.xyw(1, 5, 2));
-        builder.add(cleanUpSuperscrips, cc.xyw(1, 6, 2));
-        builder.add(cleanUpDOI, cc.xyw(1, 7, 2));
-        builder.add(cleanUpMonth, cc.xyw(1, 8, 2));
-        builder.add(cleanUpPageNumbers, cc.xyw(1, 9, 2));
-        builder.add(cleanUpUpgradeExternalLinks, cc.xyw(1, 10, 2));
-        builder.add(cleanUpMakePathsRelative, cc.xyw(1, 11, 2));
-        builder.add(cleanUpRenamePDF, cc.xyw(1, 12, 2));
+        FormBuilder builder = FormBuilder.create().layout(layout);
+        builder.add(cleanUpHTML).xyw(1, 1, 2);
+        builder.add(cleanUpUnicode).xyw(1, 2, 2);
+        builder.add(cleanUpCase).xyw(1, 3, 2);
+        builder.add(cleanUpLaTeX).xyw(1, 4, 2);
+        builder.add(cleanUpUnits).xyw(1, 5, 2);
+        builder.add(cleanUpSuperscrips).xyw(1, 6, 2);
+        builder.add(cleanUpDOI).xyw(1, 7, 2);
+        builder.add(cleanUpMonth).xyw(1, 8, 2);
+        builder.add(cleanUpPageNumbers).xyw(1, 9, 2);
+        builder.add(cleanUpUpgradeExternalLinks).xyw(1, 10, 2);
+        builder.add(cleanUpMakePathsRelative).xyw(1, 11, 2);
+        builder.add(cleanUpRenamePDF).xyw(1, 12, 2);
         String currentPattern = Globals.lang("File name format pattern").concat(": ").concat(Globals.prefs.get(ImportSettingsTab.PREF_IMPORT_FILENAMEPATTERN));
-        builder.add(new JLabel(currentPattern), cc.xyw(2, 13, 1));
-        builder.add(cleanUpRenamePDFonlyRelativePaths, cc.xyw(2, 14, 1));
-        builder.add(cleanUpBiblatex, cc.xyw(1, 15, 2));
+        builder.add(new JLabel(currentPattern)).xyw(2, 13, 1);
+        builder.add(cleanUpRenamePDFonlyRelativePaths).xyw(2, 14, 1);
+        builder.add(cleanUpBiblatex).xyw(1, 15, 2);
+	optionsPanel = builder.build();
     }
 
     private void retrieveSettings() {
