--- jabref-3.6/src/main/java/osx/macadapter/MacAdapter.java.orig
+++ jabref-3.6/src/main/java/osx/macadapter/MacAdapter.java
@@ -27,7 +27,7 @@
 import com.apple.eawt.AppEvent.QuitEvent;
 import com.apple.eawt.AppReOpenedListener;
 import com.apple.eawt.Application;
-import com.apple.eawt.FullScreenUtilities;
+//import com.apple.eawt.FullScreenUtilities;
 import com.apple.eawt.OpenFilesHandler;
 import com.apple.eawt.PreferencesHandler;
 import com.apple.eawt.QuitHandler;
@@ -44,7 +44,7 @@
         Application.getApplication().setPreferencesHandler(this);
         Application.getApplication().setQuitHandler(this);
         Application.getApplication().addAppEventListener(this);
-        FullScreenUtilities.setWindowCanFullScreen(parentFrame, true);
+//        FullScreenUtilities.setWindowCanFullScreen(parentFrame, true);
     }
 
     @Override
