--- jabref-3.8.2/src/main/java/net/sf/jabref/gui/JabRefFrame.java.osx
+++ jabref-3.8.2/src/main/java/net/sf/jabref/gui/JabRefFrame.java
@@ -143,7 +143,7 @@
 import com.jgoodies.looks.Options;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
-import osx.macadapter.MacAdapter;
+//import osx.macadapter.MacAdapter;
 
 /**
  * The main window of the application.
@@ -642,13 +642,13 @@
         //Note: The registration of Apple event is at the end of initialization, because
         //if the events happen too early (ie when the window is not initialized yet), the
         //opened (double-clicked) documents are not displayed.
-        if (OS.OS_X) {
-            try {
-                new MacAdapter().registerMacEvents(this);
-            } catch (Exception e) {
-                LOGGER.fatal("Could not interface with Mac OS X methods.", e);
-            }
-        }
+//      if (OS.OS_X) {
+//          try {
+//              new MacAdapter().registerMacEvents(this);
+//          } catch (Exception e) {
+//              LOGGER.fatal("Could not interface with Mac OS X methods.", e);
+//          }
+//      }
 
     }
 

